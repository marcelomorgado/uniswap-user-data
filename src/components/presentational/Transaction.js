import React from "react";
import PropTypes from "prop-types";

function EtherTransaction({ transaction }) {
  const { txHash, type, tokenAddress, user, amount } = transaction;
  return (
    <div>
      {txHash} - {type} - {tokenAddress} - {user} - {amount}
    </div>
  );
}
EtherTransaction.propTypes = {
  transaction: PropTypes.object.isRequired,
};

function TokenTransaction({ transaction }) {
  const { txHash, type, tokenSymbol, user, amount } = transaction;
  return (
    <div>
      {txHash} - {type} - {tokenSymbol} - {user} - {amount}
    </div>
  );
}
TokenTransaction.propTypes = {
  transaction: PropTypes.object.isRequired,
};

function Transaction({ transaction, style }) {
  const { tokenAddress } = transaction;
  const isEtherTransfer = tokenAddress === "0x";
  return (
    <div style={style}>
      {isEtherTransfer ? (
        <EtherTransaction transaction={transaction} />
      ) : (
        <TokenTransaction transaction={transaction} />
      )}
    </div>
  );
}

Transaction.propTypes = {
  transaction: PropTypes.object.isRequired,
  style: PropTypes.object.isRequired,
};

export default Transaction;
