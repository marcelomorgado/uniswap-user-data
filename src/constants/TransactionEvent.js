export const TokenPurchase = "TokenPurchase";
export const EthPurchase = "EthPurchase";
export const AddLiquidity = "AddLiquidity";
export const RemoveLiquidity = "RemoveLiquidity";

export default {
  TokenPurchase,
  EthPurchase,
  AddLiquidity,
  RemoveLiquidity,
};
