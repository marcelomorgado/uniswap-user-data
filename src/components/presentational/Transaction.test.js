import React from "react";
import { shallow } from "enzyme";
import Transaction from "./Transaction";
import TransactionEvent from "../../constants/TransactionEvent";
const { EthPurchase } = TransactionEvent;

it("renders the component", () => {
  const transaction = {
    txHash: "txHash",
    type: EthPurchase,
    tokenAddress: "0x",
    user: "0x123User",
    amount: "1",
  };
  const props = {
    transaction,
    style: {},
  };

  expect(shallow(<Transaction {...props} />)).toMatchSnapshot();
});
