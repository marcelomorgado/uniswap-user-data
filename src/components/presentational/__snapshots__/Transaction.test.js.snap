// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  style={Object {}}
>
  <div>
    txHash
     - 
    EthPurchase
     - 
    0x
     - 
    0x123User
     - 
    1
  </div>
</div>
`;
